sdk/core/src/request.rs

```diff
@@ -83,3 +127,94 @@ pub(crate) struct Parts {
     pub method: Method,
     pub headers: HeaderMap,
 }
+
+use serde::de::Visitor;
+use serde::{Deserialize, Deserializer};
+use std::collections::HashMap;
+use std::str::FromStr;
+
+impl<'de> Deserialize<'de> for Request {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_struct("Request", FIELDS, RequestVisitor)
+    }
+}
+
+struct RequestVisitor;
```

Comment:
It might be easier to deserialize to an intermediary struct that matches the on-disk represenation exactly and then perform any transformations we need when going from the intermediary struct to the final `Request` struct.

Reply:
I write out some code of what I mean here above for `BytesResponse`